pyccel/ast/bitwise_operators.py

```diff
@@ -78,25 +79,30 @@ def _set_dtype(self):
         e.g.
             1 + 2j -> PyccelAdd(LiteralInteger, LiteralComplex) -> complex
         """
-        integers  = [a for a in self._args if a.dtype in (NativeInteger(),NativeBool())]
-        reals     = [a for a in self._args if a.dtype is NativeReal()]
-        complexes = [a for a in self._args if a.dtype is NativeComplex()]
-        strs      = [a for a in self._args if a.dtype is NativeString()]
+        integers  = [a for a in _args if a.dtype in (NativeInteger(),NativeBool())]
+        reals     = [a for a in _args if a.dtype is NativeReal()]
+        complexes = [a for a in _args if a.dtype is NativeComplex()]
+        strs      = [a for a in _args if a.dtype is NativeString()]
 
         if strs or complexes or reals:
-            raise TypeError('unsupported operand type(s): {}'.format(self))
+            raise TypeError('unsupported operand type(s): {}'.format(_args))
         elif integers:
-            self._handle_integer_type(integers)
+            return self._handle_integer_type(integers)
         else:
-            raise TypeError('cannot determine the type of {}'.format(self))
+            raise TypeError('cannot determine the type of {}'.format(_args))
+
+    @staticmethod
+    def _calculate_shape_rank(*_args):
+        _rank = 0
+        _shape = ()
+        return _shape, _rank
 
-    def _set_shape_rank(self):
-        pass
 
     def _handle_integer_type(self, integers):
-        self._dtype     = NativeInteger()
-        self._precision = max(a.precision for a in integers)
+        _dtype     = NativeInteger()
+        _precision = max(a.precision for a in integers)
         self._args = [PythonInt(a) if a.dtype is NativeBool() else a for a in integers]
+        return _dtype, _precision
 
 #==============================================================================
```

Comment:
I think you missed `PyccelBitComparisonOperator`

Reply:
right, I did it on [9742d51](https://github.com/pyccel/pyccel/pull/797/commits/9742d511bef41e1cb61d7f57ba7340b159527d1d)